src/github.com/getlantern/lantern-mobile/app/src/main/java/org/getlantern/lantern/model/Utils.java

```diff
@@ -34,6 +49,53 @@ public static void clearPreferences(Context context) {
         }
     }
 
+    public static void hideKeyboard(Context context, View view) {
+        InputMethodManager inputMethodManager = (InputMethodManager)context.getSystemService(Activity.INPUT_METHOD_SERVICE);
+        inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(), 0);
+    }
+
+    public static void showErrorDialog(final FragmentActivity activity, String error) {
+        DialogFragment fragment = ErrorDialogFragment.newInstance(R.string.validation_errors, error);
+        fragment.show(activity.getSupportFragmentManager(), "error");
+    }
+
+    public static boolean isEmailValid(String email) {
+        boolean isValid = false;
+
+        String expression = "^[\\w\\.-]+@([\\w\\-]+\\.)+[A-Z]{2,4}$";
+        CharSequence inputStr = email;
+
+        Pattern pattern = Pattern.compile(expression, Pattern.CASE_INSENSITIVE);
+        Matcher matcher = pattern.matcher(inputStr);
```

Comment:
Also you can just feed email in here without creating inputStr since strings are immutable in java.